### PR TITLE
fix: activate superadmin's all vfolder listing feature

### PR DESCRIPTION
**Pull request by SonMinWoo**
resolve: https://github.com/lablup/backend.ai/issues/321
related: https://github.com/lablup/backend.ai-manager/pull/544

**When I tested, the function is working.
Trying with the admin accesskey, I get an error. Trying with the folder owner accesskey it succeeds.**

issue321_1

**But in webui, superadmin can't know all vfolders. This is because superadmin's all folder view function did not work.**
It was already implemented in the backend. One parameter required by the backend was missing, so add it.

image

It works well. (The two folders in the picture below are folders created by different accounts as shown in the picture above.)
image

To reduce the load, it request only one-time when superadmin check delegated session start checkbox.
issue321-2

### CODE REVIEW

File: src/components/backend-ai-resource-broker.ts [hunk 1]
```diff
@@ -363,11 +363,11 @@ export default class BackendAiResourceBroker extends BackendAIPage {
    * Update virtual folder list. Also divide automount folders from general ones.
    *
    */
-  async updateVirtualFolderList() {
+   async updateVirtualFolderList(userEmail = false) {
```
Comment (lizable) [inline src/components/backend-ai-resource-broker.ts]: The type of default value of parameter `userEmail` doesn't seem to have a match with SDK function in [here](https://github.com/lablup/backend.ai-webui/pull/1237/files#diff-6ab6a3b84b5af19510d01f3a5c155018416e21b94b7090c2a4a22e0a1d68e33bR1421-R1429).
It's string, not boolean.

File: src/components/backend-ai-session-launcher.ts [hunk 2]
```diff
@@ -2284,6 +2284,20 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
     }
   }
 
+  async _fetchDelegatedSessionVfolder() {
+    const ownerEnabled = this.shadowRoot.querySelector('#owner-enabled');
+    const userEmail = this.shadowRoot.querySelector('#owner-email').value;
+    if (ownerEnabled && ownerEnabled.checked) {
+      await this.resourceBroker.updateVirtualFolderList(userEmail).then(() => {
```
Comment (lizable) [inline src/components/backend-ai-session-launcher.ts]: Any reason for using `await` keyword and `.then()` keyword at the same time?

File: src/components/backend-ai-session-launcher.ts [hunk 3]
```diff
@@ -3064,6 +3078,55 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
                 `}
             </div>
+              <wl-expansion name="ownership" style="--expansion-content-padding:15px 0;">
+              <span slot="title">${_t('session.launcher.SetSessionOwner')}</span>
+              <div class="vertical layout">
+                <div class="horizontal center layout">
+                  <mwc-textfield id="owner-email" type="email" class="flex" value=""
+                                pattern="^.+@.+\..+$" icon="mail"
+                                label="${_text('session.launcher.OwnerEmail')}" size="40"></mwc-textfield>
+                  <mwc-icon-button icon="refresh" class="blue"
```
Comment (lizable) [inline src/components/backend-ai-session-launcher.ts:3188]: If I input any user account(email) and rewrite it, it doesn't update according to the new input.

File: src/lib/backend.ai-client-node.ts [hunk 4]
```diff
@@ -1540,10 +1540,15 @@ class VFolder {
   /**
    * List Virtual folders that requested accessKey has permission to.
    */
-  async list(groupId = null) {
+  async list(groupId = null, userEmail = null) {
     let reqUrl = this.urlPrefix;
-    if (groupId) {
-      const params = {group_id: groupId};
+    if (groupId || userEmail) {
```
Comment (lizable) [inline src/lib/backend.ai-client-node.ts]: I think it's better to use condition statements separately.

**Review by lizable — CHANGES_REQUESTED** [review]
I have a few things to request. Please refer to the comments below.

**Review by lizable — APPROVED** [review]
LGTM

**Comment by lizable** [comment]
I just updated two things during delegate session creation:
- [x] Enable the session delegation checkbox only if the owner's keypair is valid.
- [x] Fetch the owner's accessible vfolder list only if the owner's keypair is valid.

**Comment by inureyes** [comment]
@lizable Great!

**Review by inureyes — APPROVED** [review]

**Pull request merged**